### ISSUE TITLE
There are too many link files created.

### ISSUE THREAD

**Issue by ren-142326** [Schniz/fnm#1323]
### windows
![Snipaste_2024-11-14_21-08-25](https://github.com/user-attachments/assets/190b3e5b-8e67-40f9-a29f-5dab3a7ab70a)

### macos
<img width="1764" alt="image" src="https://github.com/user-attachments/assets/3d2d8546-5f48-415f-aab3-5c201b00cf24">

### wsl ubuntu
![Snipaste_2024-11-14_21-07-31](https://github.com/user-attachments/assets/eb71f3a7-01c4-4ad8-aba5-e738bb9e4f04)

**Comment by ren-142326** [comment]
> 这是设计使然

That's fine, but I noticed there is no cleanup plan, for example, when I close the shell, this leads to an increasing number of link files.

**Comment by ren-142326** [comment]
> 这是设计使然

I stumbled upon such a problem while using fnm, which happens occasionally, and that's how I discovered the above link file creation problem

![Snipaste_2024-11-14_21-38-34](https://github.com/user-attachments/assets/e0894cfb-d7ab-43a3-bc41-c21dd9dfdfb2)

**Comment by Schniz** [comment]
that's unrelated tho, you don't have a node version installed when fnm is bypassed.
symlinks are practically free, i still don't have a good hack in mind to clean them up. feel free to do that from time to time :)

**Comment by ren-142326** [comment]
> that's unrelated tho, you don't have a node version installed when fnm is bypassed. symlinks are practically free, i still don't have a good hack in mind to clean them up. feel free to do that from time to time :)

The fact is, I have the node version installed on my system, and a system environment variable points to the path to the fnm alias default. This is useful when switching to System, and fnm successfully takes over the installed version of System.

If I didn't install node, the shell would say no node installed and apt install node, instead of error can't find /run/user/*** node, this error is accidental, not every time.

**Comment by Schniz** [comment]
yeah exactly. "bypassing" by linking to a faulty directory is a wild hack that... works. Another hack we might pull off is to make the PATH in env something like `export PATH="$MULTISHELL/bin:$PATH:$FNM_DIR/shims"` and have a `~/.fnm/shims/node` that is simply:

```bash
#!/bin/bash

echo "You don't seem to have a system Node.js installed."
exit 1
```

So you get a nicer error. Not sure it's worth it (because Windows will have to have their own files, and we'll need to create these files on `fnm env`). Might be compatible with #609. Gotta think about this deeply.

Anyway @ren-142326 this is unrelated to the topic at hand.
